### PR TITLE
[ENH]: Add support for on-the-fly template space transformation

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry for the latest changes

This PR adds support for on-the-fly template space transformation in `junifer.data.get_parcellation()` and `junifer.data.get_mask()` enabling merging of multi-space parcellation and mask. The target space is set as per the datagrabber's specification and all of the other assets are transformed to it and then applied.

### CODE REVIEW

**Comment by synchon** [comment]
Needs to go after #243

**Review by fraimondo — DISMISSED** [review]

**Comment by synchon** [comment]
@fraimondo Can we merge this?

File: junifer/data/template_spaces.py [hunk 1]
```diff
@@ -121,15 +124,19 @@ def get_template(
...
-        If template in the required resolution is not found.
+        If required template is not found.
 
     """
     # Check for invalid space; early check to raise proper error
     if space not in tflow.templates():
         raise_error(f"Unknown template space: {space}")
 
+    # Check for template type
+    if template_type not in ["T1w", "whole", "gm", "wm", "csf"]:
```
Comment (fraimondo) [inline junifer/data/template_spaces.py:137]: What do you mean with `"whole"`. Does it come from any toolbox? I know that usually it is named "brain" mask to reference the GM, WM and CSF masks.

**Comment by fraimondo** [comment]
@synchon Just that one question.

File: junifer/data/template_spaces.py (hunk 1 above)
Comment (fraimondo) [reply junifer/data/template_spaces.py:137]: it should be "brain" then. That's the brain mask.

Comment (synchon) [reply junifer/data/template_spaces.py]: Addressed.

**Pull request merged**